db/blip_handler.go

```diff
@@ -1067,41 +1083,63 @@ func (bsc *BlipSyncContext) incrementSerialNumber() uint64 {
 	return atomic.AddUint64(&bsc.handlerSerialNumber, 1)
 }
 
-func (bsc *BlipSyncContext) addAllowedAttachments(attDigests []string) {
-	if len(attDigests) == 0 {
+func (bsc *BlipSyncContext) addAllowedAttachments(docID string, attMeta []AttachmentStorageMeta) {
+	if len(attMeta) == 0 {
 		return
 	}
 
 	bsc.lock.Lock()
 	defer bsc.lock.Unlock()
 
 	if bsc.allowedAttachments == nil {
-		bsc.allowedAttachments = make(map[string]int, 100)
+		bsc.allowedAttachments = make(map[string]AllowedAttachment, 100)
 	}
-	for _, digest := range attDigests {
-		bsc.allowedAttachments[digest] = bsc.allowedAttachments[digest] + 1
+	for _, attachment := range attMeta {
+		key := attachment.digest
+		if bsc.blipContext.ActiveProtocol() == BlipCBMobileReplicationV3 {
+			key = docID + attachment.digest
```

Comment:
Again, by moving the protocol check up one level, you can basically always have key be `docID + attachment.digest`, even when `docID` is unset.

Reply:
Sure, that make sense.